src/components/ui/CoralButton.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component,  PropTypes } from 'react';
+import Radium from 'radium';
+import color from 'color';
+
+const buildCustomStyle = (customColor) => ({
+  backgroundColor: customColor,
+  color: '#FFFFFF',
+  ':hover': {
+    backgroundColor: color(customColor).lighten(0.2).hexString()
+  },
+  active: {
+    backgroundColor: '#FFFFFF',
+    color: customColor,
+    ':hover': {
+      backgroundColor: '#FFFFFF'
+    }
+  }
+})
+
+@Radium
+class Button extends Component {
+  render() {
+    const { type = 'default', className, onClick, disabled, children, icon, style, active = false, customColor, ...rest } = this.props;
+
+    if (type === 'custom' && customColor) {
+      styles.custom = buildCustomStyle(customColor)
+    }
+
+    const finalStyles = {
+      ...styles.base,
+      ...styles[type],
+      ...style
+    };
+
+    return (
+    <button
+      className={`mdl-button mdl-js-button mdl-button--raised ${className}`}
+      onClick={onClick}
+      style={[
+        finalStyles,
+        { ':hover': { backgroundColor: color(finalStyles.backgroundColor).lighten(0.2).hexString() }},
+        active ? styles[type].active : {}
+      ]}
+      disabled={disabled ? 'disabled' : ''}
+      { ...rest }
+    >
+      { icon ? <i className="material-icons" style={styles.icon} > {icon} </i> : null }
```

Comment:
should we add a `propTypes` definition here for icon? It's a neat feature that you can pass either a component or a string here, but it's not really clear from the code.